pyesbulk/__init__.py

```diff
@@ -81,14 +118,67 @@ def quiet_loggers():
     A convenience function to quiet the urllib3 and elasticsearch1 loggers.
     """
     logging.getLogger("urllib3").setLevel(logging.FATAL)
-    logging.getLogger(_es_logger).setLevel(logging.FATAL)
+    logging.getLogger("elasticsearch").setLevel(logging.FATAL)
 
 
 # The 5xx codes on which template PUT operations are retried.
 _RETRY_5xxS = [500, 503, 504]
 
 
-def put_template(es, name=None, mapping_name=None, body=None):
+def _get_meta_version(name, body):
+    """
+    _get_meta_version Try to find an "_meta":{"version": "value"} in the
+    specified template body.
+
+    For V7 and beyond, a template should not have a name, which means the
+    mapping body includes an "_meta" key directly.
+
+    For V6 and V7 there may be an "_doc" key in the top level dict, which
+    will have "_meta" as a second level key.
+
+    Earlier versions will normally have a "document type" as the first
+    level key, with "_meta" underneath.
+
+    We're being accommodating here: if the first level dict includes an
+    "_meta" key, we'll use it. If not, and the first level dict has a single
+    key, we'll look for "_meta" under that.
+
+    Args:
+        body (dict): An Elasticsearch index document template "mappings"
+        document.
+    """
+    if not isinstance(body, dict):
+        raise TypeError
+    underbody = body
+    version = None
+    try:
+        if "_meta" not in body:
+            # Elasticsearch V1 allows multiple named templates in an index;
+            # we require that the versions be the same since we can return
+            # only one.
+            underbody = body[next(iter(body))]
+            for key in body:
+                v = int(body[key]["_meta"]["version"])
+                if not version:
+                    version = v
+                else:
+                    if v != version:
+                        raise Exception(
+                            f"Bad template, {name}: multiple templates with "
+                            "differing versions at {key}"
+                        )
+        else:
+            version = int(underbody["_meta"]["version"])
+    except KeyError:
+        raise Exception(
```

Comment:
Perhaps in another commit we could define an official `Exception` class for this and make it part of the interface?